### PR TITLE
Improve golden ticket module with hash check, better loot storage

### PR DESCRIPTION
**Pull request by k0pak4**
This draft PR partially addresses [Issue 16376](https://github.com/rapid7/metasploit-framework/issues/16376).

The changes requested/status are:

- [x] Fail quickly when given the wrong hash type
- [x] Store the ticket as a kirbi file
- [ ] Convert and store ticket as a ccache file

The first two are complete, but I'm not sure if there's a ticket converter module/function to make the ccache file. I also wasn't sure if it's possible to use the impacket module in ruby modules or not to do the conversion?

### CODE REVIEW

File: modules/post/windows/escalate/golden_ticket.rb [hunk 1]
```diff
@@ -124,14 +130,26 @@ def run
...
                                    session,
-                                   ticket,
+                                   kirbi_ticket,
                                    "#{domain}\\#{user}-golden_ticket.kirbi",
                                    "#{domain}\\#{user} Golden Ticket")
 
-      print_status("Ticket saved to #{ticket_location}")
+      print_status("Ticket saved to #{kirbi_location}")
+
+      # TODO: Convert ticket to ccache
```
Comment (smcintyre-r7) [inline modules/post/windows/escalate/golden_ticket.rb:143]: If this isn't a CCache file, then it shouldn't be stored with the `.ccache` extension. The kirbi data is already stored in the kirbi format, so this is redundant and should be removed.

Comment (k0pak4) [reply modules/post/windows/escalate/golden_ticket.rb:143]: I agree. I'm hoping to get the conversion before marking this PR as ready for review but if I don't, I'll remove this.

**Comment by k0pak4** [comment]
It looks like there's currently work being done on converting kirbi to ccache in [PR 17045](https://github.com/rapid7/metasploit-framework/pull/17045). Once that completes, I should be able to use that to store the kirbi file as a ccache file using the methods created there

**Comment by k0pak4** [comment]
@smcintyre-r7 I closed this for now. I'm going to redo these changes targeting the feature branch and will open a new PR with all three changes

**Pull request closed without merging**